doc/float.rb

```diff
@@ -0,0 +1,106 @@
+# A \Float object stores a real number
+# using the native architecture's double-precision floating-point representation.
+#
+# == \Float Imprecisions
+#
+# Some real numbers can be represented precisely as \Float objects:
+#
+#   Float(37.5)   #=> 37.5
+#   Float(98.6)   #=> 98.6
+#   Float(12.2345)#=> 12.2345
```

Comment:
I'm confused by this example. It's converting a float to a float using the `Float` method (i.e. doing nothing).

Reply:
`98.6` and `12.2345` in the example can't be represented precisely.
```ruby
'%0.20f' % 98.6 #=> "98.59999999999999431566"
'%0.20f' % 12.2345 #=> "12.23450000000000059686"
```
How about using `98.75` and `12.3125` instead? These numbers does not have such problem.

Reply:
> I'm confused by this example. It's converting a float to a float using the `Float` method (i.e. doing nothing).

Fixed.